course_catalog/etl/see.py

```diff
@@ -0,0 +1,274 @@
+    )
+    # Start and end dates in different months, same year (Jun 18 - Jul 18, 2020)
+    pattern_1_year = re.compile(
+        r"(?P<start_m>\w+)\s+(?P<start_d>\d+)\s*\-\s*(?P<end_m>\w+)\s+(?P<end_d>\d+),\s*(?P<year>\d{4})$"
+    )
+    # Start and end dates in different years (Dec 21, 2020-Jan 10,2021)
+    pattern_2_years = re.compile(
+        r"(?P<start_m>\w+)\s+(?P<start_d>\d+),\s*(?P<start_y>\d{4})\s*-\s*(?P<end_m>\w+)\s+(?P<end_d>\d+),\s*(?P<end_y>\d{4})$"
+    )
+    date_strings = [
+        rundate.strip() for rundate in course.findAll("li")[3].get_text().split("|")
+    ]
+    runs = []
+    for date_string in date_strings:
+        match = re.match(pattern_1_month, date_string)
+        if match:
+            start_date = datetime.strptime(
+                f"{match.group('start_m')} {match.group('start_d')} {match.group('year')}",
+                "%b %d %Y",
+            )
+            end_date = datetime.strptime(
+                f"{match.group('start_m')} {match.group('end_d')} {match.group('year')}",
+                "%b %d %Y",
+            )
+            runs.append(
+                (start_date.replace(tzinfo=pytz.utc), end_date.replace(tzinfo=pytz.utc))
+            )
+            continue
+        match = re.match(pattern_1_year, date_string)
+        if match:
+            start_date = datetime.strptime(
+                f"{match.group('start_m')} {match.group('start_d')} {match.group('year')}",
+                "%b %d %Y",
+            )
+            end_date = datetime.strptime(
+                f"{match.group('end_m')} {match.group('end_d')} {match.group('year')}",
+                "%b %d %Y",
+            )
+            runs.append(
+                (start_date.replace(tzinfo=pytz.utc), end_date.replace(tzinfo=pytz.utc))
+            )
+            continue
+        match = re.match(pattern_2_years, date_string)
+        if match:
+            start_date = datetime.strptime(
+                f"{match.group('start_m')} {match.group('start_d')} {match.group('start_y')}",
+                "%b %d %Y",
+            )
+            end_date = datetime.strptime(
+                f"{match.group('end_m')} {match.group('end_d')} {match.group('end_y')}",
+                "%b %d %Y",
+            )
+            runs.append(
+                (start_date.replace(tzinfo=pytz.utc), end_date.replace(tzinfo=pytz.utc))
+            )
+    return runs
+
+
+def _parse_instructors(details):
+    """
+    Extract instructor names from the course detail page
+
+    Args:
+        details(Tag): BeautifulSoup Tag for course details
+
+    Returns:
+        list of dict: List of first & last names of each instructor
+
+    """
+    instructors = [
+        instructor.find("h2").get_text().split()
+        for instructor in details.findAll("article", {"class": "faculty-article"})
+    ]
+    return [
+        {"first_name": instructor[0], "last_name": instructor[1]}
+        for instructor in instructors
```

Comment:
Is it possible to rewrite as `{"first_name", first_name, "last_name", last_name} for first_name, last_name in instructors}`? Or are there more items in each tuple?